### PR TITLE
Use github action matrix in CI

### PR DESCRIPTION
**Pull request by brokenpip3**
* Split the original job in 4 different one
* Bats-assert and bats-file need the bats-support job to be correctly terminated (see `needs`)
* In bats-assert and bats-file we don't rely on brew automatic dependency install, instead we still install bats-support from local Formula file in PR branch. The reason behind this is that in case of future bats-support release we can test both the bats-support build and the file/assert tests with that new bats-support version.
* Fix misspelling: `Formulae` not `Formulas`

In case of approval please squash commits before merging this PR

### CODE REVIEW

**Comment by martin-schulze-vireso** [comment]
I think we can considerably reduce the amount of repetition by using https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix the only difference seems to be preinstalling bats-support, but this step could be [made conditional](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idstepsif).

> In bats-assert and bats-file we don't rely on brew automatic dependency install, instead we still install bats-support from local Formula file in PR branch. The reason behind this is that in case of future bats-support release we can test both the bats-support build and the file/assert tests with that new bats-support version.

I am not sure if I got that right. Are you saying this should be some kind of early out test for the dependency version upgrade before it gets published to homebrew?

**Comment by brokenpip3** [comment]
Let me refresh this PR so we can also bump the bats-assert version.

**Comment by brokenpip3** [comment]
@martin-schulze-vireso now should be like your suggestion (`bats-file` is failing but that's another story).
After merging this PR I will create a new one to bump the `bats-assert` version

**Comment by brokenpip3** [comment]
@martin-schulze-vireso I also fixed the `bats-file` checks by adding another condition (if bats-file) while installing libs that depends on support.. like file but I forgot about that in the last iteration here.
Could you please review/approve?

**Review by martin-schulze-vireso — APPROVED** [review]

**Comment by martin-schulze-vireso** [comment]
Seeing that you have been the most active on this repo, would you like to join the maintainer ranks for it?

**Comment by brokenpip3** [comment]
> @brokenpip3 Seeing that you have been the most active on this repo, would you like to join the maintainer ranks for it?

yep absolutely, I will be happy to do it :)

**Comment by martin-schulze-vireso** [comment]
I'm done for today, I hope I can onboard you in my next busy cycle.

**Comment by brokenpip3** [comment]
Thanks for the invite, will try to do my best :) 
(also will try to join the chat)

**Pull request merged**
